packages/wasm-dpp/src/document/state_transition/document_batch_transition/document_transition/document_create_transition.rs

```diff
@@ -25,39 +37,211 @@ impl From<DocumentCreateTransition> for DocumentCreateTransitionWasm {
 
 #[wasm_bindgen(js_class=DocumentCreateTransition)]
 impl DocumentCreateTransitionWasm {
+    #[wasm_bindgen(constructor)]
+    pub fn from_raw_object(
+        raw_object: JsValue,
+        data_contract: &DataContractWasm,
+    ) -> Result<DocumentCreateTransitionWasm, JsValue> {
+        let data_contract: DataContract = data_contract.clone().into();
+        let mut value = raw_object.with_serde_to_json_value()?;
+        let document_type = value
+            .get_string(document::property_names::DOCUMENT_TYPE)
+            .with_js_error()?;
+
+        let (identifier_paths, _) = data_contract
+            .get_identifiers_and_binary_paths(document_type)
+            .with_js_error()?;
+        // Allow to fail as it could be a Buffer or Identifier
+        let _ = value.replace_identifier_paths(
+            identifier_paths
+                .into_iter()
+                .chain(document_create_transition::IDENTIFIER_FIELDS),
+            ReplaceWith::Bytes,
+        );
+        let transition =
+            DocumentCreateTransition::from_raw_object(value, data_contract).with_js_error()?;
+
+        Ok(transition.into())
+    }
+
+    // DocumentCreateTransition
+    #[wasm_bindgen(js_name=getEntropy)]
+    pub fn entropy(&self) -> Vec<u8> {
+        self.inner.entropy.to_vec()
+    }
+
+    #[wasm_bindgen(js_name=getCreatedAt)]
+    pub fn created_at(&self) -> Option<i64> {
+        self.inner.created_at
+    }
+
+    #[wasm_bindgen(js_name=getUpdatedAt)]
+    pub fn updated_at(&self) -> Option<i64> {
+        self.inner.updated_at
+    }
+
+    #[wasm_bindgen(js_name=getRevision)]
+    pub fn revision(&self) -> u32 {
+        document_transition::INITIAL_REVISION
+    }
+
+    // AbstractDocumentTransitionMethods
+    #[wasm_bindgen(js_name=getId)]
+    pub fn id(&self) -> IdentifierWrapper {
+        self.inner.base.id.clone().into()
+    }
+
+    #[wasm_bindgen(js_name=getType)]
+    pub fn document_type(&self) -> String {
+        self.inner.base.document_type.clone()
+    }
+
     #[wasm_bindgen(js_name=getAction)]
     pub fn action(&self) -> u8 {
         self.inner.base.action as u8
     }
 
+    #[wasm_bindgen(js_name=getDataContract)]
+    pub fn data_contract(&self) -> DataContractWasm {
+        self.inner.base.data_contract.clone().into()
+    }
+
+    #[wasm_bindgen(js_name=getDataContractId)]
+    pub fn data_contract_id(&self) -> IdentifierWrapper {
+        self.inner.base.data_contract.id.clone().into()
+    }
+
+    #[wasm_bindgen(js_name=get)]
```

Comment:
I think we don't need to write js_name if we're not renaming anything, just #[wasm_bindgen] :)

Reply:
fixed